sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -560,6 +585,249 @@ public String toJson() {
         }
     }
 
+    /**
+     * Gets the UserAgent header value used by the client issueing this operation
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
+     * a custom {@link CosmosDiagnosticsHandler}
+     * @return the UserAgent header value used for the client that issued this operation
+     */
+    public String getUserAgent() {
+        if (this.diagnostics == null) {
+            return "";
+        }
+
+        CosmosDiagnostics diagnostics = this.diagnostics.peekFirst();
+        if (diagnostics == null) {
+            return "";
+        }
+
+        return diagnostics.getUserAgent();
+    }
+
+    /**
+     * Returns the set of contacted regions
```

Comment:
FYI - incorrect java doc - 

```suggestion
     * Returns the connection mode used in the client.
```